server/asset/eth/eth_test.go

```diff
@@ -225,31 +212,27 @@ func TestDecodeCoinID(t *testing.T) {
 }
 
 func TestRun(t *testing.T) {
-	// TODO: Test all paths.
 	ctx, cancel := context.WithCancel(context.Background())
-	header1 := &types.Header{Number: big.NewInt(1)}
-	block1 := types.NewBlockWithHeader(header1)
-	blockHash1 := block1.Hash()
-	node := &testNode{
-		bestBlkHash: blockHash1,
-		blk:         block1,
-	}
-
 	backend := unconnectedETH(tLogger, new(config))
 	ch := backend.BlockChannel(1)
+	node := &testNode{
+		bestHdr: &types.Header{Number: big.NewInt(0)},
+	}
 	backend.node = node
+	backend.hashCache.prime(ctx, node)
+	*node = testNode{
+		bestHdr: &types.Header{
+			Number: big.NewInt(1),
+		},
+	}
```

Comment:
Why not `node.bestHdr = &types.Header{Number: big.NewInt(1)}`?

Reply:
yeah, that probably makes more sense

Reply:
Changed.